main.js

```diff
@@ -0,0 +1,273 @@
+class WeatherApp {
+  constructor() {
+    this.APIKEY = '16806dd9a591b25a5beebeb69dd718b8';
+    this.coordinates = {};
+    this.weatherInfo = {};
+    this._init();
+  }
+
+  _init() {
+    const searchForm = document.querySelector('.weather__search-form');
+    searchForm.addEventListener('submit', () => this._checkLocation(event));
+  }
+
+  async _checkLocation(event) {
+    event.preventDefault();
+    const locationInput = document.querySelector('.weather__textbox').value;
+
+    if (!locationInput) {
+      return;
+    }
+
+    await this._getCoordinates(locationInput);
+  }
+
+  async _getCoordinates(location) {
+    try {
+      this._startLoading();
+      const coordResponse = await fetch(`http://api.openweathermap.org/geo/1.0/direct?q=${location}&limit=10&appid=${this.APIKEY}`);
+
+      // TODO:400系ごとのハンドリングを分ける
+      if (!coordResponse.ok) {
+        this._throwError(coordResponse.status);
+      }
+
+      const coordArr = await coordResponse.json();
+
+      if (coordArr.length === 0) {
+        const noCityHtml = 'お探しの都市の天気情報が見つかりませんでした。<br>別の都市名を入力し、再度お試しください。'
+        this._displayError(noCityHtml);
+        return;
+      }
+
+      if (coordArr.length >= 2) {
+        this._getMultiLocationNames(coordArr);
+        return;
+      }
+
+      this.coordinates = {
+        lat: coordArr[0].lat,
+        lon: coordArr[0].lon,
+      }
+
+      await this._getWeather(this.coordinates);
+
+    } catch (e) {
+      this._catchError(e);
+
+    } finally {
+      this._endLoading();
+    }
+
+  }
+
+  _getMultiLocationNames(coordArr) {
+    const multiLoc = coordArr.map(location => {
+      //都市の日本語名が表示可能な場合は日本語名の表示を優先するための分岐
+      if (location.local_names && location.local_names.ja) {
+        return this._extractLocInfo(location.local_names.ja, location);
+
+      } else {
+        return this._extractLocInfo(location.name, location);
+      }
+    });
+
+    const multiLocMap = Array.from(new Map(multiLoc.map(obj => [obj.name, obj])));
+    const multiLocUni = multiLocMap.map(location => location[1]);
+
+    if (multiLocUni.length === 1) {
+      this._getWeather(multiLocUni);
+
+    } else {
+      this._renderMultiLoc(multiLocUni);
+    }
+  }
+
+  _extractLocInfo(locName, locInfo) {
```

Comment:
[nits]
こちらの関数ですが、関数としての内容が薄すぎますね
location渡してそのなかでextractしてもらいましょうよ
extractってなまえなんだから

```suggestion
  _extractLocInfo(location) {
      if (location.local_names.ja) {
        // 日本語都市名
      }
      else if (location.location_names) {
          // 日本語じゃない都市名
      }
      else {
          // location
      }
    
  }
```

Reply:
@cyndiii29 
条件分岐や必要情報の抽出を_extractLocInfo内にまとめました。
引数としてlocationのみを渡すように改修しています。